core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -156,47 +165,96 @@ object ConsumerGroupCommand {
-        zkUtils.readDataMaybeNull(groupDirs.consumerOwnerDir + "/" + topicPartition.partition)._1.map { owner =>
-          topicPartition -> owner
+      val topicPartitions = getTopicsPartitions(topics)
+      var groupMemberIds = zkUtils.getConsumersInGroup(group)
+
+      // mapping of topic partition -> ephemeral owner
+      val eOwnerByTopicPartition = mutable.Map[TopicAndPartition, Long]()
+      topicPartitions.foreach { topicPartition =>
+        val data = zkUtils.readDataMaybeNull(new ZKGroupTopicDirs(group, topicPartition.topic).consumerOwnerDir + "/" + topicPartition.partition)
+        val owner = data._1
+        if (!owner.isEmpty) {
+          val eOwner = data._2.getEphemeralOwner()
+          eOwnerByTopicPartition.put(topicPartition, eOwner)
         }
+      }
+
+      // mapping of ephemeral owner -> group member id
+      val memberIdByEOwner = groupMemberIds.flatMap { memberId =>
+        val eOwner = zkUtils.readDataMaybeNull(new ZKGroupDirs(group).consumerGroupDir + "/ids/" + memberId)._2.getEphemeralOwner()
+        Some(eOwner).map { eOwner => eOwner -> memberId }
       }.toMap
-      val partitionOffsets = getPartitionOffsets(group, topicPartitions, channelSocketTimeoutMs, channelRetryBackoffMs)
-      describeTopicPartition(group, topicPartitions, partitionOffsets.get, ownerByTopicPartition.get)
+
+      // mapping of topic partition -> member id
+      val memberIdByTopicPartition = eOwnerByTopicPartition.flatMap { case (topicPartition, eOwner) =>
+        val memberId = memberIdByEOwner.getOrElse(eOwner, "")
+        Some(memberId.toString()).map { memberId => topicPartition -> memberId }
+      }.toMap
+
+      // mapping of member id -> list of topic partitions
+      val memberTopicPartitions = memberIdByTopicPartition groupBy{_._2} map {
+        case (key, value) => (key, value.unzip._1.toArray) }
+
+      // mapping of member id -> list of subscribed topics
+      val topicsByMemberId = zkUtils.getTopicsPerMemberId(group, true)
+
+      topicPartitions.foreach { topicPartition =>
+        val partitionOffsets = getPartitionOffsets(group, List(topicPartition), channelSocketTimeoutMs, channelRetryBackoffMs)
+        val memberId = memberIdByTopicPartition.get(topicPartition)
+        // since member id is repeated in owner, leave owner and owner host empty
+        describeMemberTopicPartition(group, topicPartition.topic, List(topicPartition), partitionOffsets.get, _ => memberId, _ => None, _ => None)
+        if (!memberId.isEmpty) {
+          groupMemberIds = groupMemberIds.filterNot { _ == memberId.get }
+        }
+      }
+
+      groupMemberIds.sortBy { - memberTopicPartitions.get(_).size }.foreach { memberId =>
+        topicsByMemberId(memberId).foreach { topic =>
+          val topicPartitions = memberTopicPartitions.getOrElse(memberId, Array()).filter { _.topic == topic }
+          val partitionOffsets = getPartitionOffsets(group, topicPartitions, channelSocketTimeoutMs, channelRetryBackoffMs)
```

Comment:
This is the case for members not owning any topic partitions, right? If so, could we just set topicPartitions and partitionOffsets to empty?